src/views/transaction-details/components/chain-row/chain-row.tsx

```diff
@@ -0,0 +1,40 @@
+import { FC } from "react";
+
+import { ReactComponent as EthChainIcon } from "src/assets/icons/ethereum.svg";
+import { ReactComponent as HermezChainIcon } from "src/assets/icons/polygon-hermez.svg";
+import Typography from "src/views/shared/typography/typography.view";
+import useChainRowStyles from "src/views/transaction-details/components/chain-row/chain-row.style";
+
+interface ChainRowProps {
+  type: "l1" | "l2";
+  side: "to" | "from";
+  className?: string;
+}
+
+const ChainRow: FC<ChainRowProps> = ({ type, side, className }) => {
+  const classes = useChainRowStyles();
+
+  const chain = {
+    l1: { from: "hermez", to: "eth" },
+    l2: { from: "eth", to: "hermez" },
+  };
+
+  switch (chain[type][side]) {
+    case "eth":
+      return (
+        <Typography type="body1" className={className}>
+          <EthChainIcon /> Ethereum chain
+        </Typography>
+      );
+    case "hermez":
+      return (
+        <Typography type="body1" className={className}>
+          <HermezChainIcon className={classes.hermezChain} /> Polygon Hermez chain
+        </Typography>
+      );
+    default:
+      return null;
```

Comment:
This feels like a [smell](https://martinfowler.com/bliki/CodeSmell.html). We should not need this default.

Reply:
Typescript doesn't like it 
`Type 'Element | undefined' is not assignable to type 'ReactElement<any, any> | null'.`

Reply:
That is the point. Typescript should like what we do. The fact that we need this default value to make Typescript shut up indicates a flaw in type design. We need to change the design.

Reply:
When you create this demo data:

```typescript
const demoData = [{
    id: Math.random(),
    type: "l1",
    token: "dai",
    timestamp: Date.now() - Math.floor(Math.random() * 10000),
    status: "on-hold",
    amount: Math.floor(Math.random() * 100),
}]
```
I interpret `type` as the target chain. Is this correct?. This `type` name prop is ambiguous. I'd change it to `target` in our domain to be more clear.

With a clearer naming, we no longer need to pass both `type` and `side` to this component. If I understand the current logic correctly:

```bash
When ("to" & "l1") | ("from" & "l2") we want to render Ethereum
When ("to" & "l2") | ("from" & "l1") we want to render Polygon
```

Why not have a dummy component like this:

```typescript
interface ChainRowProps {
  targetChain: "l1" | "l2";
  className?: string;
}

const ChainRow: FC<ChainRowProps> = ({ targetChain: targetChain, className }) => {
  const classes = useChainRowStyles();

  const EthereumChain = (
    <Typography type="body1" className={className}>
      <EthChainIcon /> Ethereum chain
    </Typography>
  );

  const PolygonHermezChain = (
    <Typography type="body1" className={className}>
      <HermezChainIcon className={classes.hermezChain} /> Polygon Hermez chain
    </Typography>
  );

  return targetChain === "l1" ? EthereumChain : PolygonHermezChain;
};
```

And move the decision making to the instantiator:

```typescript
type TargetChain = "l1" | "l2";

// this assumes `type` as `target`
const { amount, token, timestamp, type, status } = data;
const toTargetChain: TargetChain = type;
const fromTargetChain: TargetChain = type === "l2" ? "l1" : "l2";

....


<div className={classes.row}>
  <Typography type="body2" className={classes.alignRow}>
    From
  </Typography>
  <ChainRow targetChain={fromTargetChain} className={classes.alignRow} />
</div>
<div className={classes.row}>
  <Typography type="body2" className={classes.alignRow}>
    To
  </Typography>
  <ChainRow targetChain={toTargetChain} className={classes.alignRow} />
</div>
```

Reply:
Fixed.